easybuild/easyblocks/generic/systemmpi.py

```diff
@@ -0,0 +1,192 @@
+                       mpi_name, self.mpi_version, self.mpi_prefix)
+
+        # For the version of the underlying C compiler need to explicitly extract (to be certain)
+        self.compiler_version, debug_msg = extract_compiler_version(self.mpi_c_compiler)
+        self.log.debug(debug_msg)
+        self.log.debug("Derived compiler/version for C compiler underneath system MPI %s: %s, %s",
+                       mpi_name, self.mpi_c_compiler, self.compiler_version)
+
+        # If EasyConfig specified "real" version (not 'system' which means 'derive automatically'), check it
+        if self.cfg['version'] == 'system':
+            self.log.info("Found specified version '%s', going with derived compiler version '%s'",
+                          self.cfg['version'], self.mpi_version)
+        elif self.cfg['version'] != self.mpi_version:
+            raise EasyBuildError("Specified version (%s) does not match version reported by MPI (%s)" %
+                                 (self.cfg['version'], self.mpi_version))
+
+        # fix installdir and module names (may differ because of changes to version)
+        mns = ActiveMNS()
+        self.cfg.full_mod_name = mns.det_full_module_name(self.cfg)
+        self.cfg.short_mod_name = mns.det_short_module_name(self.cfg)
+        self.cfg.mod_subdir = mns.det_module_subdir(self.cfg)
+
+        # keep track of original values, for restoring later
+        self.orig_version = self.cfg['version']
+        self.orig_installdir = self.installdir
+
+    def make_installdir(self, dontcreate=None):
+        """Custom implementation of make installdir: do nothing, do not touch system compiler directories and files."""
+        pass
+
+    def make_module_req_guess(self):
+        """
+        A dictionary of possible directories to look for.  Return empty dict for a system compiler.
+        """
+        return {}
+
+    def make_module_step(self, fake=False):
+        """
+        Custom module step for SystemMPI: make 'EBROOT' and 'EBVERSION' reflect actual system MPI version
+        and install path.
+        """
+        # First let's verify that the toolchain and the compilers under MPI match
+        c_compiler_name = self.toolchain.COMPILER_CC
+        compiler_version = get_software_version(self.toolchain.COMPILER_MODULE_NAME[0])
+
+        if self.mpi_c_compiler != c_compiler_name and self.compiler_version != compiler_version:
+            raise EasyBuildError("C compiler for toolchain (%s/%s) and underneath MPI (%s/%s) do not match!",
+                                 c_compiler_name, compiler_version, self.mpi_c_compiler, self.compiler_version)
+
+        # For module file generation: temporarily set version and installdir to system compiler values
+        self.cfg['version'] = self.mpi_version
+        self.installdir = self.mpi_prefix
+
+        # Generate module
+        res = super(SystemMPI, self).make_module_step(fake=fake)
```

Comment:
There's a bug here but I cannot find it. In the final module it is setting
```
setenv("EBVERSIONOPENMPI","5.4.0")
```
where `5.4.0` is the version of the compiler underneath MPI. I've checked the value of `self.mpi_version` so I don't understand why this value is winning...is it somehow connected to `altversion`?